### DISCUSSION TITLE
ESPHOME - Sonoff che accende ciclicalmente luci tuya

### DISCUSSION THREAD

**Discussion by antoniopanella** [indomus/forum#1345] [Hardware & Firmware]
Salve a tutti, sto provando a far accendere due luci tuya (aimè collegate in cloud per il momento) tramite un sonoff mini collegato a un gruppo pulsanti. Attualmente il sonoff mini da tensione alle 2 luci tuya che si accendono entrambe, se voglio spegnerne una do il comando voce con amazon echo (accendi salotto: accende il mini, spegni ingresso: spegne luce ingresso, spegni portono: spegne luce ingresso). Quindi armato di santa pazienza ho flashato un bios su un mini a banco e replicato la situazione. 

Il mio primo obiettivo è che tenendo premuto il pulsante del sonoff per 3 secondi si accenda e spenga la tensione alle 2 lampade e questo non mi ha dato problemi.

Il mio secondo obiettivo è quello di far ciclare le accensioni luci a ogni click del pulsante s1-s2 sul sonoff. 

Con un primo yaml sono riuscito a fare funzionare il tutto ma ce la latenza del segnale che arriva dal cloud verso home assistant e quindi se clicco di continuo il pulsante non ottengo l'effetto rotazione voluto.
Ecco il primo yaml https://pastebin.com/HZMhzsdH

Con un secondo yaml ho cercato di creare una variabile locale all'esphome per cambiare lo stato al click localmente.
Ho provato varie cose quella piu vicina al risultato è stato il template switch, che ho sfruttato per cachare lo stato di luce ingresso e portone e per mandare i comandi ad home assistant.
Purtroppo cosi facendo ho ottenuto 2 problemi:
a) il binary_sensor sembra non ricevere piu lo stato delle luci da home assistant
b) se cambio lo stato dei 2 switch template, apparentemente lo fa ma al secondo click lo stato rimane precedente (vedi i log)
Yaml seconda soluzione: https://pastebin.com/mYrw8xkH
Log: https://pastebin.com/etihYAYY

Ultima domanda per gli esperti di yaml:
Nelle if di controllo di stato avevo provato a testare se i 2 switch fossero off per dare una sequenza diversa (tipo accendo entrambi, poi 1, poi l'altro e poi spenfo), ma non mi è stato possibile.
Ho problemi sia nella condition che nello spegnimento.
Se nella condtion metto 
```
- if:
    condition:
      and:
        switch.is_off: sw_portone
	    switch.is_off: sw_ingresso

```
Ricevo l'errore che il secondo comando è duplicato
Stessa cosa nella then per accendere entrambe
```
then:
    switch.turn_on: sw_portone
    switch.turn_on: sw_ingresso
```
	
Scusate il lungo posto, un grazie a chi saprà darmi suggerimenti utili

**Comment by Nik71git** [comment]
Ciao, scusa la domanda: riesci a postare un jpg di come hai fatto i collegamenti elettrici? Da quello che intuisco è come se tu avessi messo tre interruttori in serie....
